ow_plexil/src/plans/ActionGoalStateDemo.plp

```diff
@@ -2,57 +2,78 @@
-// is being investigated.
+// Pan and tilt the antenna repeatedly, while monitoring and printing
+// action goal state. Looks for just the four supported states.
+// Terminate this plan by injecting a power fault: Low
+// State-Of-Charge, Instantaneous Capacity Loss, or Thermal Failure.
 
 #include "ow-interface.h"
 
+LibraryAction TestAntennaMovement;
+
 ActionGoalStateDemo:
 {
   log_info ("Starting ActionGoalStateDemo plan...");
-  LibraryCall Tilt (Degrees=0);
 
   Run: Concurrence
   {
-    Real NewAngle = 0;
-    ExitCondition Lookup(PowerFault);
+    Real PanAngle = 0;
+    Real TiltAngle = 0;
 
-    Stowing: UncheckedSequence
+    MoveAntenna: UncheckedSequence
     {
-      StartCondition !Lookup(ArmFault);
-      RepeatCondition true;
-      LibraryCall ArmUnstow();
-      LibraryCall ArmStow();
+      RepeatCondition !Lookup(PowerFault);
+
+      PanAngle = PanAngle + 15;
+      if (PanAngle > 180) PanAngle = PanAngle - 360; endif;
+      TiltAngle = TiltAngle + 5;
+      if (TiltAngle > 45) TiltAngle = TiltAngle - 90; endif;
+      LibraryCall Tilt (Degrees=TiltAngle);
+      Wait 1;
+      LibraryCall Pan (Degrees=PanAngle);
+      Wait 1;
     }
-    Panning: UncheckedSequence
+
+    ReportStatus: UncheckedSequence
     {
-      StartCondition !Lookup(AntennaPanFault);
-      RepeatCondition true;
+      RepeatCondition !Lookup(PowerFault);
 
-      if (Lookup(ActionGoalStatus("ArmUnstow")) == ACTION_ACTIVE) {
-        NewAngle = (Lookup(PanDegrees) + 15) mod 360;
+      if (Lookup(ActionGoalStatus("Tilt")) == ACTION_INACTIVE) {
+        log_info ("Tilt is inactive.");
       }
-      else if (Lookup(ActionGoalStatus("ArmStow")) == ACTION_ACTIVE) {
-        NewAngle = (Lookup(PanDegrees) - 15) mod 360;
+      else if (Lookup(ActionGoalStatus("Tilt")) == ACTION_ACTIVE) {
+        log_info ("Tilt is active.");
       }
-      endif;
-
-      if (NewAngle > 180) {
-        NewAngle = NewAngle - 360;
+      else if (Lookup(ActionGoalStatus("Tilt")) == ACTION_SUCCEEDED) {
+        log_info ("Tilt is successful.");
+      }
+      else if (Lookup(ActionGoalStatus("Tilt")) == ACTION_ABORTED) {
+        log_info ("Tilt is aborted.");
+      }
+      if (Lookup(ActionGoalStatus("Pan")) == ACTION_INACTIVE) {
+        log_info ("Pan is inactive.");
+      }
+      else if (Lookup(ActionGoalStatus("Pan")) == ACTION_ACTIVE) {
+        log_info ("Pan is active.");
+      }
+      else if (Lookup(ActionGoalStatus("Pan")) == ACTION_SUCCEEDED) {
+        log_info ("Pan is successful.");
+      }
+      else if (Lookup(ActionGoalStatus("Pan")) == ACTION_ABORTED) {
+        log_info ("Pan is aborted.");
+      }
+      else {
+        log_warning ("Either Pan or Tilt are in an unexpected state:");
+        log_warning ("Pan is in state ",
+                     Lookup(ActionGoalStatus("Pan")));
+        log_warning ("Tilt is in state ",
+                     Lookup(ActionGoalStatus("Tilt")));
       }
```

Comment:
It'd be cool if these statement could be triggered only upon state transition instead of performing checks at a regular rate, but I'm not sure how.

Reply:
Polling was actually the intention here -- to see what the state is at a steady interval.   Reacting just to changes could be done in the usual way: a start condition that checks if the state is different from a cached state.